src/elm/arithmetic.js

```diff
@@ -279,7 +302,12 @@ class Log extends Expression {
       return null;
     }
 
-    return args.reduce((x, y) => Math.log(x) / Math.log(y));
+    const log = args.reduce((x, y) => Math.log(x) / Math.log(y));
+
+    if (!MathUtil.isValidDecimal(log)) {
+      return null;
+    }
+    return log;
```

Comment:
This is the third time I've seen this pattern in this file.  I'd consider making a function to handle this pattern (`nanToNull` or `suppressNaN` or `decimalOrNull` or something like that).

Reply:
I added `decimalOrNull` to the `MathUtil` file and used it to replace this repetitive pattern.